tiledb/libtiledb.pyx

```diff
@@ -5427,6 +5427,19 @@ cdef class SparseArrayImpl(Array):
                     out[final_name] = arr
 
         return out
+    
+    def unique_dim_values(self, dim=None):
+        if dim and not self.domain.has_dim(dim):
+            raise ValueError("Array does not contain Dimension '{}'.".format(dim))
+
+        query = self.query(attrs=[])[:]
+
+        if dim:
+            dim_values = np.unique(query[dim])
+        else:
+            dim_values = [tuple(np.unique(values)) for values in query.values()]
```

Comment:
Seems like this should be an `OrderedDict` rather than a tuple of arrays. What do you think?

Reply:
OK that makes sense. I'll change it to be an OrderedDict of tuples.